core/src/test/java/org/apache/calcite/test/SqlToRelTestBase.java

```diff
@@ -657,6 +664,32 @@ public RelNode trimRelNode(RelNode relNode) {
       return relNode;
     }
 
+    private Pair<SqlToRelConverter, SqlValidator> createSqlToRelConverter() {
+      final RelDataTypeFactory typeFactory = getTypeFactory();
+      final Prepare.CatalogReader catalogReader =
```

Comment:
No, we better put the validator creation and converter creation in 2 methods.

Reply:
@danny0405  Dear danny,  I have made the change according to our discussion. I'm sorry to bother you on vacation